tools/exporters/DB_dump_table_data.php

```diff
@@ -54,11 +66,13 @@
  *                      values to accommodate for timezone differences.
  */
 
+
 // Loop through all tables to generate insert statements for each.
 foreach ($tableNames as $tableName) {
     $paths = \NDB_Config::singleton()->getSetting('paths');
     $filename = $paths['base'] . "/raisinbread/RB_files/RB_$tableName.sql";
-    exec('mysqldump '.$databaseInfo['database'].' '.
+    exec('mysqldump -u '.$dbUser.' -p'.$dbPassword.' -h '.$dbHost.' '.
```

Comment:
These variables need to be shell escaped. You can use https://www.php.net/manual/en/function.escapeshellarg.php

Regarding the warning: MySQL flags it as a warning because anyone running "ps" on the host will be able to see the password. To properly fix it securely, you'd probably have to use proc_open instead of exec so you can write to the user input side of the pipe from php, but that would make things significantly more complicated.

Reply:
changes done